File: app/src/main/java/com/dimatest/movieapp/App/MovieApp.java
package com.dimatest.movieapp.App;

import android.app.Application;

import com.dimatest.movieapp.di.component.AppComponent;
import com.dimatest.movieapp.di.component.DaggerAppComponent;

public class MovieApp extends Application {

    private static MovieApp app;
    private AppComponent appComponent;

    public static MovieApp getApp() {
        return app;
    }

    public AppComponent getAppComponent() {
        return appComponent;
    }

    @Override
    public void onCreate() {
        super.onCreate();
        app = this;
        appComponent = DaggerAppComponent.builder()

                .build();
        appComponent.inject(this);
    }
}

File: app/src/main/java/com/dimatest/movieapp/app/MovieApp.kt
package com.dimatest.movieapp.app

import android.app.Application
import com.dimatest.movieapp.di.modules.*
import org.koin.android.ext.koin.androidContext
import org.koin.android.ext.koin.androidLogger
import org.koin.core.context.startKoin

class MovieApp : Application() {

    override fun onCreate() {
        super.onCreate()
        initKoin()
    }
    private fun initKoin() {
        startKoin {
            androidLogger()
            androidContext(this@MovieApp)
            modules(listOf(dataBaseModule, networkModule,
                    useCaseModule, viewModelModule))
        }
    }
}